### PR TITLE
$replace modifier

### PR DESCRIPTION
**Pull request by chrmod**
requires https://github.com/ghostery/adblocker/pull/4241

To test basic use-case, try this custom filter `example.com$replace=/Example Domain/Hello from Ghostery/g` at example.com
Screenshot 2024-08-26 at 18 29 47
Notice that both `<title>` and `<h1>` got updated.

This have to be carefully tested on youtube.com once the `$replace` filters are enabled.

### CODE REVIEW

File: extension-manifest-v3/src/background/adblocker.js
Comment (philipp-classen) [inline extension-manifest-v3/src/background/adblocker.js:515]:
```suggestion
      const cspPolicies = [];
      const htmlFilters = [];
```

**Review by philipp-classen — APPROVED** [review]

**Comment by philipp-classen** [comment]
Update: For YouTube, we found a problems with the new adblocker version (`replace` regular expression can corrupt JSON requests). That needs to be addressed before we can merge.

**Comment by GRadziejewski** [comment]
## Smoke tests Iteration no 1:
Tests of: https://github.com/ghostery/ghostery-extension/pull/1842#issuecomment-2329350552

---
Checked features:

- Enable/Disable Ad-blocking 
- Enable/Disable Anti-tracking	
  - URL / Trackers blocked on UI
- Enable/Disable Never Consent  
- Pause Ghostery
- Regional filters
  - Check YT
  - Check YT Kids

```
Browser: Chrome 128.0.6613.120
OS: Windows 10
GBE 10.4.3

Browser: Firefox 130.0
OS: Windows 11
GBE 10.4.3

Browser: Firefox 130.0
OS: macOS 14.6.1
GBE 10.4.3
```
---


## Issues:
- 

## Fixed:
- 

## Cannot reproduce:
- 

## Will not be fixed:
-

**Review by GRadziejewski — APPROVED** [review]
LGTM

**Review by philipp-classen — APPROVED** [review]

File: extension-manifest-v3/src/background/adblocker.js
Comment (smalluban) [inline extension-manifest-v3/src/background/adblocker.js:522]: Shouldn't we add the same constraint on other even listeners? I found a case where we have a callback for our own background page in Firefox (Navigation API) when Dev Tools are opened. 
Sometimes I have an error in Chrome, that shows that page from `chrome://` can't be accessed.

**Pull request merged**
